proxies/shared/resources/jsc/MessageBatches.Create.Validate.js

```diff
@@ -14,154 +14,162 @@ try {
-              typeof message.recipient.dateOfBirth !== "undefined"
-              && message.recipient.dateOfBirth !== null
-              && (
-                typeof message.recipient.dateOfBirth !== "string"
-                || !dobRegex.test(message.recipient.dateOfBirth)
-              )
-            ) {
-              pushError(invalidError(pointer))
-            }
-
-              // $.data.attributes.messages.x.personalisation
-              pointer = "/data/attributes/messages/" + index + "/personalisation"
-              if (typeof message.personalisation !== "object") {
-                pushError(invalidError(pointer))
+              // $.data.attributes.messages.x.recipient
+              pointer = "/data/attributes/messages/" + index + "/recipient";
+              if (isUndefined(message.recipient)) {
+                pushError(missingError(pointer));
+              } else if (typeof message.recipient !== 'object') {
+                pushError(invalidError(pointer));
+              } else if (message.recipient === null) {
+                pushError(nullError(pointer));
+              } else {
+
+                // $.data.attributes.messages.x.recipient.nhsNumber
+                pointer = "/data/attributes/messages/" + index + "/recipient/nhsNumber";
+                if (isUndefined(message.recipient.nhsNumber)) {
+                  pushError(missingError(pointer));
+                } else if (message.recipient.nhsNumber === null) {
+                  pushError(nullError(pointer));
+                } else if (typeof message.recipient.nhsNumber !== "string" || !nhsNumberRegex.test(message.recipient.nhsNumber)) {
+                  pushError(invalidError(pointer));
+                }
+
+                // $.data.attributes.recipients.x.dateOfBirth
+                pointer = "/data/attributes/messages/" + index + "/recipient/dateOfBirth";
+                if (
+                  !isUndefined(message.recipient.dateOfBirth)
+                  && message.recipient.dateOfBirth !== null
+                  && (
+                    typeof message.recipient.dateOfBirth !== "string"
+                    || !dobRegex.test(message.recipient.dateOfBirth)
+                  )
+                ) {
+                  pushError(invalidError(pointer));
+                }
+
+                // $.data.attributes.messages.x.personalisation
+                pointer = "/data/attributes/messages/" + index + "/personalisation";
+                if (typeof message.personalisation !== "object") {
+                  pushError(invalidError(pointer));
+                }
               }
             }
-          }
-        })
+          });
+        }
       }
     }
+  } catch (err) {
+    console.log(err);
   }
 }
 
-
 function pushError(error) {
   // Limit the amount of errors returned to 100 entries
   if (errors.length <= 100) {
-    errors.push(error)
+    errors.push(error);
+  } else {
+    throw new Error("Maximum number of error entries reached.");
```

Comment:
Exceptions can have quite a big overhead- we can probably add a check for each message instead which just terminates the for each loop instead.

If you can show that the exception doesn't add a massive overhead then I'll accept this!